source/common/http/http2/codec_impl.cc

```diff
@@ -349,6 +354,36 @@ void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMapVector& metadat
   }
 }
 
+void ConnectionImpl::StreamImpl::processBufferedData(bool stream_being_destroyed) {
+  ENVOY_CONN_LOG(debug, "Stream {} processing buffered data.", parent_.connection_, stream_id_);
+
+  auto next_stage = stream_manager_.getNextStage();
+  stream_manager_.flush_all_data_ = stream_being_destroyed;
```

Comment:
maybe some comments?  on first pass my I wonder why we'd flush all data if the stream is going away?  seems like a lot of work for a dying stream

Reply:
I guess I'm confused how anything is closing the stream if the codec doesn't communicate to other layers that the stream is complete?

Reply:
As discussed offline, ended up achieving this by buffering onStreamClose call that way we don't have this hack where we flush all the data through.